src/Uno.UI.RemoteControl.Host/RemoteControlExtensions.cs

```diff
@@ -1,118 +1,219 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Diagnostics;
-using System.Globalization;
 using System.Threading;
 using Microsoft.AspNetCore.Builder;
-using Microsoft.AspNetCore.Routing;
 using Microsoft.Extensions.Configuration;
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
 using Uno.Extensions;
-using Uno.UI.RemoteControl.Server.Helpers;
 using Uno.UI.RemoteControl.Services;
 using Uno.UI.RemoteControl.Server.Telemetry;
+using Uno.UI.RemoteControl.Server.AppLaunch;
+using Microsoft.AspNetCore.Http;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Uno.UI.RemoteControl.VS.Helpers;
+using System.Text.Json;
 
 namespace Uno.UI.RemoteControl.Host
 {
-	static class RemoteControlExtensions
+	internal static class RemoteControlExtensions
 	{
-		public static IApplicationBuilder UseRemoteControlServer(
-			this IApplicationBuilder app,
+		public static WebApplication UseRemoteControlServer(
+			this WebApplication app,
 			RemoteControlOptions options)
 		{
-			app.UseRouter(router =>
-			{
-				router.MapGet("rc", async context =>
-				{
-					if (!context.WebSockets.IsWebSocketRequest)
+			app
+				.MapGet(
+					"/rc",
+					async context =>
 					{
-						context.Response.StatusCode = 400;
-						return;
-					}
+						await HandleWebSocketConnectionRequest(app, context);
+					}).WithName("RC Protocol");
 
-					if (app.Log().IsEnabled(LogLevel.Information))
-					{
-						app.Log().LogInformation($"Accepted connection from {context.Connection.RemoteIpAddress}");
-					}
+			// HTTP GET endpoint to register an app launch
+			app.MapGet(
```

Comment:
As we are not updated to latest feature, why not create a `Controller` ? (Especially since there are a lot of new method below)